pkg/aws/policies.go

```diff
@@ -1975,3 +1989,78 @@ func (c *awsClient) listRoleAttachedPolicies(roleName string) ([]iamtypes.Attach
 
 	return attachedPoliciesOutput.AttachedPolicies, nil
 }
+
+// check whether the policy contains specified tags
+func isPolicyTagFit(c client.IamApiClient, poilcyArn *string, tagFilter map[string]string) (bool, error) {
+	if len(tagFilter) != 0 {
+		tags, err := c.ListPolicyTags(context.Background(),
+			&iam.ListPolicyTagsInput{
+				PolicyArn: poilcyArn,
+			},
+		)
+		if err != nil {
+			return false, err
+		}
+		fitTagSize := 0
+		for _, tag := range tags.Tags {
+			value, ok := tagFilter[aws.ToString(tag.Key)]
+			if ok && value != aws.ToString(tag.Value) {
+				return false, nil
+			}
+			fitTagSize++
+		}
+		if fitTagSize < len(tagFilter) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func getAttachedPolicies(c client.IamApiClient, role string,
+	tagFilter map[string]string) ([]string, error) {
+	policyArr := []string{}
+	policiesOutput, err := c.ListAttachedRolePolicies(context.Background(),
+		&iam.ListAttachedRolePoliciesInput{
+			RoleName: aws.String(role),
+		})
+	if err != nil && !awserr.IsNoSuchEntityException(err) {
```

Comment:
Do we need to check the exception here?
What happens if policy doesn't exist?
I feel a nil ptr error might occur at:
`for _, policy := range policiesOutput.AttachedPolicies`?

Reply:
oh, I copy this from existing code. Thanks for finding this problem. fixed it